lisa/tools/cargo.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from pathlib import PurePath
+from typing import List, Optional, Type, cast
+
+from lisa.executable import Tool
+from lisa.operating_system import CBLMariner, Posix, Ubuntu
+from lisa.tools.curl import Curl
+from lisa.tools.echo import Echo
+from lisa.tools.gcc import Gcc
+from lisa.tools.ln import Ln
+from lisa.util import LisaException
+from lisa.util.process import ExecutableResult
+
+
+class Cargo(Tool):
+    @property
+    def command(self) -> str:
+        return "cargo"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Gcc]
+
+    def _install(self) -> bool:
+        os = self.node.os
+        cargo_source_url = "https://sh.rustup.rs"
+        try:
+            if isinstance(os, CBLMariner) or isinstance(os, Ubuntu):
+                self.__install_dependencies()
+
+                # install cargo/rust
+                curl = self.node.tools[Curl]
+                result = curl.fetch(
+                    arg="-sSf",
+                    url=cargo_source_url,
+                    execute_arg="-s -- -y",
+                    shell=True,
+                )
+                result.assert_exit_code()
+
+                echo = self.node.tools[Echo]
+                home_dir = echo.run(
+                    "$HOME",
+                    shell=True,
+                    expected_exit_code=0,
+                    expected_exit_code_failure_message="failure to grab $PATH via echo",
+                ).stdout
+
+                ln = self.node.tools[Ln]
+                ln.create_link(
+                    is_symbolic=True,
+                    target=f"{home_dir}/.cargo/bin/cargo",
+                    link="/usr/local/bin/cargo",
+                )
+            else:
+                raise LisaException(
+                    f"os '{os.name}' doesn't support cargo installation."
+                )
```

Comment:
this still is caught by below except, we need throw UnsupportedDistroException out, and catch it in test case, in except mark them as skip

Reply:
Sure Lili, will do that way